collection/collection.go

```diff
@@ -75,7 +270,7 @@ var (
 // persist collections, serializing to a directory of "profileID.json" files,
 // with one for each profileID in the set of collections. providing an empty
 // repoDir value will create an in-memory collection
-func NewLocalSet(ctx context.Context, bus event.Bus, repoDir string, options ...LocalSetOptionFunc) (Set, error) {
+func NewLocalSet(ctx context.Context, repoDir string, options ...LocalSetOptionFunc) (WritableSet, error) {
```

Comment:
I am a fan of removing the Bus, since this implementation doesn't need it, and returning a WritableSet to avoid the need for a type cast later on.